File: yapc-kansai/example/perl/polymorphism.pl
package Animal {
    sub new {
        my $class = shift;
        bless { @_ }, $class;
    }
    sub say { print shift->{say}."\n" }
};
package Dog {
    @ISA = qw/Animal/;
    sub new { bless Animal->SUPER::new(say => "Waon!!"), shift }
};
package Cat {
    @ISA = qw/Animal/;
    sub new { bless Animal->SUPER::new(say => "Nyan!!"), shift }
};
my ($dog, $cat) = (Dog->new, Cat->new);
$dog->say();
$cat->say();
print $dog->isa('Animal'), ", ", $cat->isa('Animal');

File: yapc-kansai/example/go/polymorphism.go
package main

import "fmt"

type Animal interface {
	Say()
}
type Dog string
type Cat string

func (Dog) Say()        { fmt.Println("Waon!!") }
func (Cat) Say()        { fmt.Println("Nyan!!") }
func Say(animal Animal) { animal.Say() }

func main() {
	var dog Dog
	var cat Cat
	Say(dog)
	Say(cat)
}
